### ISSUE TITLE
SyntaxError: Missing export in esm@3.0.35

### ISSUE THREAD

**Issue by clarkdo** [standard-things/esm#411]
Error:
> SyntaxError: Missing export 'createTest2' in ES module: node_modules/test/build.js
>     at Object.<anonymous> (xxx.js:1)

Reproducable structure:
1. esm 
```js
require = require('esm')(module)
requireModule('./xxx')
```
2. xxx.js import `node_modules/test`
```js
import { createTest2 } from 'test' 
```
3. `node_modules/test/index.js`

```js
module.exports = require('./build')
```

4. `node_modules/test/build.js`

```js
exports.createTest1 = function () {};
exports.createTest2 = function () {};
```

Example:
https://github.com/vuejs/vue/blob/dev/packages/vue-server-renderer/index.js#L16
https://github.com/vuejs/vue/blob/dev/packages/vue-server-renderer/build.js#L8641-L8642

**Comment by jdalton** [comment]
Hi @clarkdo could you create a simple repro repo.

**Comment by jdalton** [comment]
Thanks @clarkdo!

This appears to be fixed in our master branch :tada:

**Comment by clarkdo** [comment]
@jdalton Thank you so much for the fast fix 😸

**Comment by StreetStrider** [comment]
@jdalton looks like I've faced that or similar issue on `3.0.36` with package `randomstring`.
`SyntaxError: Missing export 'generate' in ES module: file:///.../node_modules/randomstring/lib/randomstring.js`.
When I do `import { generate as random } from 'randomstring'`. (possible caused by [this re-export](https://github.com/klughammer/node-randomstring/blob/f504e8586d9ada54ca2642eb1752d49f7f48acde/index.js))

I can believe that this may be an improper use of `import` by mine, but this works previously on major `esm@3` til recent time.

**Comment by jdalton** [comment]
@StreetStrider it's likely totally my bad _(`esm` loader regression)_ and not something you've done. I should have a release out today that hopefully resolves the issue.

**Comment by jdalton** [comment]
**Update:**

[v3.0.37](https://github.com/standard-things/esm/releases/tag/3.0.37) is released.

**Comment by StreetStrider** [comment]
Sorry, I've tried `3.0.37` when its out, havent any time to post this. Now esm nagging on `export { default } from './Outcome'` with `Missing export 'default' in ES module: file:///.../src/Outcome/index.js` where static typing doesn't find any problem one step earlier. I use this when I want re-export `Smth/Smth.js` as `Smth/index.js`.

**Comment by StreetStrider** [comment]
Hard fixed with `"esm": "3.0.34",` for now.

**Comment by jdalton** [comment]
HI @StreetStrider!

Could you create a repro repo for me to check out and file a new issue.

**Comment by StreetStrider** [comment]
@jdalton will do.

**Comment by StreetStrider** [comment]
Confirm that fixed in #422 :+1:
